### ISSUE TITLE
muti-gpus train support

### ISSUE THREAD

**Issue by TalonX1** [filaPro/oneformer3d#3]
Thank you for your excellent work. May I ask if the code repository supports multi GPU training?

**Comment by filaPro** [comment]
Sure, if we are inherited from mmdetection3d you can follow multi-gpu [guide](https://mmsegmentation.readthedocs.io/en/latest/user_guides/4_train_test.html#training-and-testing-on-multiple-gpus-and-multiple-machines) of mmengine. The only thing, you probably need just to copy `tools/dist_train.py` script from mmdetection3d to our repo.

**Comment by TalonX1** [comment]
> Sure, if we are inherited from mmdetection3d you can follow multi-gpu [guide](https://mmsegmentation.readthedocs.io/en/latest/user_guides/4_train_test.html#training-and-testing-on-multiple-gpus-and-multiple-machines) of mmengine. The only thing, you probably need just to copy `tools/dist_train.py` script from mmdetection3d to our repo.


Nice,thank you!

**Comment by RayYoh** [comment]
Hi, everyone. When I use dist_train, if I directly use Ctrl+C to terminate the program, I always got zombie processes. It means there is no PID but the memory of GPU is occupied. @filaPro @TalonX1

**Comment by filaPro** [comment]
Looks like an issue of mmdetection3d, not oneformer3d. Is the situation the same when you run any config directly from their repo?

**Comment by RayYoh** [comment]
> Looks like an issue of mmdetection3d, not oneformer3d. Is the situation the same when you run any config directly from their repo?

Yes, I also think it is an issue of mmdetection3d. I have resolved it by finding the parent process. But I am confused about the reason.

**Comment by tabbabb** [comment]
Hello, I am currently training a model on the S3DIS dataset using multiple GPUs with the dist_train.sh script. However, I encountered an issue with torch.nn.parallel.DistributedDataParallel. The error message indicates that the reduction was not finished in the prior iteration, and some parameters did not receive gradients. How should I do?

**Comment by yangfy0608** [comment]
> Hello, I am currently training a model on the S3DIS dataset using multiple GPUs with the dist_train.sh script. However, I encountered an issue with torch.nn.parallel.DistributedDataParallel. The error message indicates that the reduction was not finished in the prior iteration, and some parameters did not receive gradients. How should I do?

I met the same error, have you addressed the problem?

**Comment by col14m** [comment]
> > Hello, I am currently training a model on the S3DIS dataset using multiple GPUs with the dist_train.sh script. However, I encountered an issue with torch.nn.parallel.DistributedDataParallel. The error message indicates that the reduction was not finished in the prior iteration, and some parameters did not receive gradients. How should I do?
> 
> I met the same error, have you addressed the problem?

Have you tried setting `find_unused_parameters = True` in your config?

**Comment by yangfy0608** [comment]
> > > Hello, I am currently training a model on the S3DIS dataset using multiple GPUs with the dist_train.sh script. However, I encountered an issue with torch.nn.parallel.DistributedDataParallel. The error message indicates that the reduction was not finished in the prior iteration, and some parameters did not receive gradients. How should I do?
> > 
> > 
> > I met the same error, have you addressed the problem?
> 
> Have you tried setting `find_unused_parameters = True` in your config?

Yes, i have tried yet, and it works! Thanks for your reply!
Another question is when I try to use mutil_card from 1gpu*batchsize=16 to 8gpu*batchsize=2, is it necessary to change the learning rate?

**Comment by col14m** [comment]
> > > > Hello, I am currently training a model on the S3DIS dataset using multiple GPUs with the dist_train.sh script. However, I encountered an issue with torch.nn.parallel.DistributedDataParallel. The error message indicates that the reduction was not finished in the prior iteration, and some parameters did not receive gradients. How should I do?
> > > 
> > > 
> > > I met the same error, have you addressed the problem?
> > 
> > 
> > Have you tried setting `find_unused_parameters = True` in your config?
> 
> Yes, i have tried yet, and it works! Thanks for your reply! Another question is when I try to use mutil_gpu from 1gpu with batchsize=16 to 8gpu with batchsize=2, is it necessary to change the learning rate?

I don't know the correct answer, I couldn't establish any dependencies between the learning rate and the number of GPUs. My advice is to check everything empirically :)

**Comment by yangfy0608** [comment]
> > > > > Hello, I am currently training a model on the S3DIS dataset using multiple GPUs with the dist_train.sh script. However, I encountered an issue with torch.nn.parallel.DistributedDataParallel. The error message indicates that the reduction was not finished in the prior iteration, and some parameters did not receive gradients. How should I do?
> > > > 
> > > > 
> > > > I met the same error, have you addressed the problem?
> > > 
> > > 
> > > Have you tried setting `find_unused_parameters = True` in your config?
> > 
> > 
> > Yes, i have tried yet, and it works! Thanks for your reply! Another question is when I try to use mutil_gpu from 1gpu with batchsize=16 to 8gpu with batchsize=2, is it necessary to change the learning rate?
> 
> I don't know the correct answer, I couldn't establish any dependencies between the learning rate and the number of GPUs. My advice is to check everything empirically :)

Fine! Thanks for your help!
